libs/utils/src/common/common.utils.ts

```diff
@@ -66,3 +67,58 @@ export function validateEnvironmentVariables(logger?: any): void {
     );
   }
 }
+
+const logTransformer =
+  (logger: Logger) =>
+  (type: string, ...args: any[]) => {
+    const message = args
+      .map((arg) =>
+        typeof arg === 'string'
+          ? arg
+          : // safely serialize objects / errors
+            (() => {
+              try {
+                return JSON.stringify(arg);
+              } catch {
+                return String(arg);
+              }
+            })(),
+      )
+      .join(' ');
+
+    // now call the original, so it still goes to stderr
+    logger[type](message);
+  };
+
+export function setupLoggingOverrides() {
```

Comment:
Moved the `BigInt` monkey patch here. Also monkey patch `console.log`, etc, because polkadot-js has a number of console logs that bypass the logger.